File: aircon/src/main/java/com/ironsource/aura/aircon/JsonConverter.java
package com.ironsource.aura.aircon;

import java.lang.reflect.Type;

public interface JsonConverter {

	String toJson(Object obj);

	<T> T fromJson(String json, Class<T> clazz) throws
			JsonException;

	<T> T fromJson(String json, Type type) throws
			JsonException;

	class JsonException
			extends Exception {

		public JsonException() {
		}

		public JsonException(final String message) {
			super(message);
		}

		public JsonException(final String message, final Throwable cause) {
			super(message, cause);
		}

		public JsonException(final Throwable cause) {
			super(cause);
		}
	}
}

File: aircon/src/main/java/com/ironsource/aura/aircon/JsonConverter.kt
package com.ironsource.aura.aircon

import java.lang.reflect.Type

/**
 * Created on 9/5/2019.
 */
interface JsonConverter {

    fun toJson(obj: Any): String

    @Throws(JsonException::class)
    fun <T> fromJson(json: String, clazz: Class<T>): T?

    @Throws(JsonException::class)
    fun <T> fromJson(json: String, type: Type): T?

    class JsonException : Exception {

        constructor()

        constructor(message: String) : super(message)

        constructor(message: String, cause: Throwable) : super(message, cause)

        constructor(cause: Throwable) : super(cause)
    }
}